backend/layers/processing/process_download.py

```diff
@@ -75,37 +91,106 @@ def remove_prefix(string: str, prefix: str) -> str:
         else:
             return string
 
-    def process(self, dataset_id: DatasetVersionId, dataset_uri: str, artifact_bucket: str, sfn_task_token: str):
+    @staticmethod
+    def get_job_definion_name(dataset_vid: str) -> str:
+        if os.getenv("REMOTE_DEV_PREFIX"):
+            stack_name = os.environ["REMOTE_DEV_PREFIX"].replace("/", "")
+            prefix = f"{os.environ['DEPLOYMENT_STAGE']}-{stack_name}"
+        else:
+            prefix = f"{os.environ['DEPLOYMENT_STAGE']}"
+        job_definition_name = f"dp-{prefix}-ingest-process-{dataset_vid}"
+        return job_definition_name
+
+    @staticmethod
+    def estimate_resource_requirements(
+        adata: scanpy.AnnData,
+        memory_modifier: float = MEMORY_MODIFIER,
+        min_memory_MB: int = MIN_MEMORY_MB,
+        min_vcpu: int = MIN_VCPU,
+        max_memory_MB: int = MAX_MEMORY_MB,
+        max_vcpu: int = MAX_VCPU,
+        swap_memory_MB: int = SWAP_MEMORY_MB,
+        memory_per_vcpu: int = MEMORY_PER_VCPU,
+    ) -> Dict[str, int]:
+        """
+        Estimate the resource requirements for a given dataset
+
+        :param adata: The datasets AnnData object
+        :param memory_modifier: A multiplier to increase/decrease the memory requirements by
+        :param min_memory_MB: The minimum amount of memory to allocate.
+        :param min_vcpu: The minimum number of vCPUs to allocate.
+        :param max_memory_MB: The maximum amount of memory to allocate.
+        :param max_vcpu: The maximum number of vCPUs to allocate.
+        :param memory_per_vcpu: The amount of memory to allocate per vCPU.
+        :param swap_memory_MB:
+        :return: A dictionary containing the resource requirements
+        """
+        # Note: this is a rough estimate of the uncompressed size of the dataset. This method avoid loading the entire
+        # dataset into memory.
+        uncompressed_size_MB = adata.n_obs * adata.n_vars / MB
+        estimated_memory_MB = max([uncompressed_size_MB * memory_modifier, min_memory_MB])
+        if estimated_memory_MB > max_memory_MB:
+            estimated_memory_MB = max_memory_MB
+            estimated_vcpus = max_vcpu
+            max_swap = swap_memory_MB
+        else:
+            estimated_vcpus = max([estimated_memory_MB / memory_per_vcpu, min_vcpu])
+            max_swap = 0
+
+        return {"Vcpus": int(ceil(estimated_vcpus)), "Memory": int(ceil(estimated_memory_MB)), "MaxSwap": max_swap}
```

Comment:
move the `int(ceil())` logic up to ln 137, where you are actually calculating the `estimated_vcpus`. That var should always hold a whole number.

Reply:
I was trying to avoid overloading those lines with nested function calls.